packages/react-native-executorch/common/rnexecutorch/models/ocr/DetectorUtils.cpp

```diff
@@ -0,0 +1,697 @@
+ * in a presumed quadrilateral, identifies the two shortest sides, and fits a
+ * linear model to the midpoints of these sides. It also evaluates whether the
+ * resulting line should be considered vertical based on a predefined threshold
+ * for the x-coordinate differences.
+ *
+ * If the line is vertical it is fitted as a function of x = my + c, otherwise
+ * as y = mx + c.
+ *
+ * @return A tuple with 2 floats and a bool, where:
+ *   - the first float represents the slope (m) of the line.
+ *   - the second float represents the line's intercept (c) with y-axis.
+ *   - a bool indicating whether the line is
+ * considered vertical.
+ */
+std::tuple<float, float, bool>
+fitLineToShortestSides(const std::array<Point, 4> &points) {
+  std::array<std::pair<float, float>, 4> sides;
+  std::array<Point, 4> midpoints;
+#pragma unroll
+  for (std::size_t i = 0; i < midpoints.size(); i++) {
+    const auto p1 = points[i];
+    const auto p2 = points[(i + 1) % midpoints.size()];
+
+    const float sideLength = distanceFromPoint(p1, p2);
+    sides[i] = std::make_pair(sideLength, i);
+    midpoints[i] = midpointBetweenPoint(p1, p2);
+  }
+
+  // Sort the sides by length ascending
+  std::sort(sides.begin(), sides.end());
+
+  const Point midpoint1 = midpoints[sides[0].second];
+  const Point midpoint2 = midpoints[sides[1].second];
+  const float dx = std::fabs(midpoint2.x - midpoint1.x);
+
+  float m, c;
+  bool isVertical;
+
+  std::vector<cv::Point2f> cvMidPoints = {
+      cv::Point2f(midpoint1.x, midpoint1.y),
+      cv::Point2f(midpoint2.x, midpoint2.y)};
+  cv::Vec4f line;
+  // parameteres for fitLine calculation:
+  constexpr int numericalParameter =
+      0; // important only for some types of distances, O means an optimal value
+         // is chosen
+  constexpr double accuracy =
+      0.01; // sufficient accuracy. Value proposed by OPENCV
+
+  isVertical = dx < verticalLineThreshold;
+  if (isVertical) {
+    for (auto &pt : cvMidPoints) {
+      std::swap(pt.x, pt.y);
+    }
+  }
+  cv::fitLine(cvMidPoints, line, cv::DIST_L2, numericalParameter, accuracy,
+              accuracy);
+  m = line[1] / line[0];
+  c = line[3] - m * line[2];
+  return {m, c, isVertical};
+}
+
+std::array<Point, 4> rotateBox(const std::array<Point, 4> &box, float angle) {
+  const Point center = centerOfBox(box);
+
+  const float radians = angle * M_PI / 180.0f;
+
+  std::array<Point, 4> rotatedPoints;
+  for (std::size_t i = 0; i < box.size(); ++i) {
+    const Point &point = box[i];
+    const float translatedX = point.x - center.x;
+    const float translatedY = point.y - center.y;
+
+    const float rotatedX =
+        translatedX * std::cos(radians) - translatedY * std::sin(radians);
+    const float rotatedY =
+        translatedX * std::sin(radians) + translatedY * std::cos(radians);
+
+    const Point rotatedPoint = {.x = rotatedX + center.x,
+                                .y = rotatedY + center.y};
+    rotatedPoints[i] = rotatedPoint;
```

Comment:
Without loss of readability ;)
```suggestion
    rotatedPoints[i] = {.x = rotatedX + center.x,
                                .y = rotatedY + center.y};
```